impl/src/settings.rs

```diff
@@ -148,6 +149,63 @@ pub struct RazeSettings {
    */
   #[serde(default = "default_raze_settings_rust_rules_workspace_name")]
   pub rust_rules_workspace_name: String,
+
+  /**
+   * The expected path relative to the `Cargo.toml` file where vendored sources can
+   * be found. This should match the path passed to the `cargo vendor` command. eg:
+   * `cargo vendor -q --versioned-dirs "cargo/vendor"
+   */
+  #[serde(default = "default_raze_settings_vendor_dir")]
+  pub vendor_dir: String,
+}
+
+/** 
+ * Additional build files (see [crate::settings::CrateSettings::additional_build_file])
+ * are both representable by a path to a file and a name. This enum allows for this attribute
+ * to be representable by the path provided in the metadata (which is a relative path to a
+ * file from a `Cargo.toml` file) but resolvable to a canonicalized path so that the additional
+ * BUILD file may be accessed at any point. Build files that have relitive paths associated with
+ * them are considered `Validated`.
+ */
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum AdditionalBuildFile {
+  Raw(String),
+  Validated { name: String, path: PathBuf },
+}
+
+impl From<&str> for AdditionalBuildFile {
+  fn from(value: &str) -> Self {
+    AdditionalBuildFile::Raw(value.to_owned())
+  }
+}
+
+impl<'build_file> From<&'build_file AdditionalBuildFile> for &'build_file String {
```

Comment:
This seems weird and kind of magical. Why not just "to_string" or "to_debug_string".

(the problem with automagic conversion to string is that there's generally not a great canonical string representation of some type for all use cases -- hence a choice like "to_debug_string")

(Edit: I wrote this comment before the general comment on AdditionalBuildFile. I'm hoping you can scrap most of this machinery, and then this comment will be moot.)

Reply:
Gonna hold on addressing this until https://github.com/google/cargo-raze/pull/301#discussion_r538019610 is resolved